### PR TITLE
Always create dest dir so build.log is accessible

### LINKED ISSUE #51: build log not created if initial build fails
The backstory: a new `18f-pages` branch was created on https://github.com/18f/team-browser, but the build requires a certain envrionment variable to be set. The Jekyll build failed on Pages, as expected, but then https://pages.18f.gov/team-browser/build.log gave a 404. Found this error in `/var/log/pages/pages.log`:

```
Error moving build log: ENOENT: no such file or directory, open '/usr/local/18f/pages/sites/pages.18f.gov/team-browser/build.log'
```

My guess at the solution is that [`copyLog()`](https://github.com/18F/pages-server/blob/73b9874d3e3daa0301d59fcce94f26e48a8518c7/lib/site-builder.js#L172-L182) needs an [`fs.mkdir()`](https://nodejs.org/api/fs.html#fs_fs_mkdir_path_mode_callback).

### PR DESCRIPTION
**Pull request by mbland**
Closes #51.

Also note that https://nodejs.org/api/errors.html#errors_class_system_error is
currently incorrect, as demonstrated by output from a test run during the
course of developing this feature:

{ [Error: EEXIST: file already exists, mkdir '.../dest_dir/foo']
  errno: -17,
  code: 'EEXIST',
  syscall: 'mkdir',
  path: '.../dest_dir/foo'
}

cc: @afeld @jbarnicle @catherinedevlin @DavidEBest

### CODE REVIEW

File: lib/site-builder.js [hunk 1]
```diff
@@ -154,11 +154,26 @@ SiteBuilder.launchBuilder = function(info, branch, builderConfig) {
...
 
+function createBuildDestinationDir(buildDestinationDir) {
+  return new Promise(function(resolve, reject) {
+    fs.mkdir(buildDestinationDir, function(err) {
+      if (err && err.code !== 'EEXIST') {
+        return reject(new Error('Failed to create ' + buildDestinationDir +
+          ': ' + err.message));
+      }
+      resolve();
+    });
```
Comment (afeld) [inline lib/site-builder.js:174]: I'd suggest using https://github.com/substack/node-mkdirp to simplify this.

File: test/site-builder-test.js [hunk 2]
```diff
@@ -333,12 +332,28 @@ describe('SiteBuilder', function() {
...
     it('should create a builder that fails to build the site', function() {
       var handler = SiteBuilder.makeBuilderListener(webhook, builderConfig),
           errMsg = 'Error: failed to clone foo ' +
             'with exit code 1 from command: ' +
             'git clone git@github.com:18F/foo.git --branch 18f-pages';
 
+      // Ensure the output dir gets cleaned up. Though cloning the repo will
+      // fail, the builder will still create the output directory so that the
+      // build.log file will still be accessible to diagnose the error.
+      filesHelper.dirs.push(path.join('dest_dir', 'foo'));
```
Comment (afeld) [inline test/site-builder-test.js:356]: Maybe we should check that the `build.log` file exists, too?

**Comment by afeld** [comment]
1) STOP WORKING. 2) Thanks for the fast turnaround time, which apparently hasn't changed :beers:

File: lib/site-builder.js (hunk 1 above)
Comment (mbland) [inline lib/site-builder.js:174]: Not sure it simplified it that much per se, but it seems a robust and widely-used package, so done. (Also tweaked the error message slightly.)

File: test/site-builder-test.js (hunk 2 above)
Comment (mbland) [inline test/site-builder-test.js:356]: That was already happening by accident. Notice the test is already checking the contents of `dest_dir/foo/build.log`. The fixture was originally creating `dest_dir/foo` in `beforeEach`, but this PR takes that out, to make sure the new code is creating that directory instead.

**Comment by mbland** [comment]
Funny, doesn't feel like work, at least not at the moment. :stuck_out_tongue:

File: test/site-builder-test.js (hunk 2 above)
Comment (afeld) [inline test/site-builder-test.js:356]: I guess my question is: would this test fail without the corresponding change in `site-builder.js`?

Comment (mbland) [inline test/site-builder-test.js:356]: I made sure it did. That's how I always roll (or at least try to). :wink:

**Pull request merged**